addons/pos_loyalty/static/src/overrides/models/pos_store.js

```diff
@@ -215,6 +215,15 @@ patch(PosStore.prototype, {
         for (const reward of this.models["loyalty.reward"].getAll()) {
             this.compute_discount_product_ids(reward, this.models["product.product"].getAll());
         }
+
+        for (const program of this.models["loyalty.program"].getAll()) {
+            if (program.date_to) {
+                program.date_to = new Date(program.date_to);
+            }
+            if (program.date_from) {
+                program.date_from = new Date(program.date_from);
+            }
+        }
```

Comment:
i agree that this works, but why doesn't the `getAll` function return the dates in the correct format from the beginning? Maybe we can add this logic directly in the `related_models.js` file? There must be also other places in pos that are impacted by this change